wasmVirtualMachine.js

```diff
@@ -423,18 +471,85 @@ function build_module(byte_code) {
                 i++;
                 decode = Leb.decodeUint32(byte_code, i);
                 i = decode.nextIndex;
-                size = decode.value;
+                expected_end = i + decode.value;
+
+                // get array of globals
+                decode = Leb.decodeUint32(byte_code, i);
+                i = decode.nextIndex;
+                num_els = decode.value;
+
+                // get global
+                for (let j = 0; j < num_els; j++) {
+                    // get global description (type and mutability, mut is 0 or 1)
+                    decode = Leb.decodeUint32(byte_code, i);
+                    i = decode.nextIndex;
+                    let type = decode.value;
+
+                    // no need to decode since varuint1 (only one byte)
+                    let mut = byte_code[i];
+                    i++;
+
+                    // evaluate initializer expression can be either get_global or const
+                    let init_value;
+                    let ret = parse_init_expression(byte_code, i);
+                    init_value = ret.value;
+                    i = ret.nextIndex;
+
+                    // construct global
+                    globals.push(new GlobalInstance(type, init_value, mut));
+                }
+
+                // sanity check
+                if (expected_end != i) {
+                    console.log("alignment issue when parsing");
+                    return -1;
+                }
 
-                // TODO
                 break;
 
             case export_section_id:
                 i++;
                 decode = Leb.decodeUint32(byte_code, i);
                 i = decode.nextIndex;
-                size = decode.value;
+                expected_end = i + decode.value;
+
+                // get array of exports
+                decode = Leb.decodeUint32(byte_code, i);
+                i = decode.nextIndex;
+                num_els = decode.value;
+
+                // get export instance
+                for (let j = 0; j < num_els; j++) {
+                    // get name (identifier: byte array wich is valid UTF-8)
+                    let name = [];
+                    decode = Leb.decodeUint32(byte_code, i);
+                    i = decode.nextIndex;
+                    let len = decode.value;
+
+                    for (let k = 0; k < len; k++) {
+                        name.push(byte_code[i]);
+                        i++;
+                    }
+
```

Comment:
It would probably be a good idea to convert the name from a byte-array into a string by decoding the chars. The chars are encoded using utf-8
https://webassembly.github.io/spec/binary/values.html#binary-name